upgrade/upgrade-1.2.12.php

```diff
@@ -30,5 +30,41 @@
  */
 function upgrade_module_1_2_12($module)
 {
+    foreach (PrestaShop\Module\PrestashopCheckout\OrderStates::ORDER_STATES as $state => $color) {
+        $orderStateId = (int) \Configuration::get($state);
+        setStateIcons($state, $orderStateId, $module->name);
+    }
+
     return $module->registerHook('displayInvoiceLegalFreeText');
 }
+
+function setStateIcons($state, $orderStateId, $moduleName)
+{
+    $iconExtension = '.gif';
+    $iconToPaste = _PS_ORDER_STATE_IMG_DIR_ . $orderStateId . $iconExtension;
+
+    if (true === file_exists($iconToPaste)) {
```

Comment:
If file_exists fails it sends a warning exception, wouldn't it be better to do a try catch just in case to avoid an exception?

Reply:
php docs says `Upon failure, an E_WARNING is emitted.`